### PR TITLE
add test code “matplotlib-applied.ipynb”

### PR DESCRIPTION
**Pull request by Zhaochengxiann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 1]
```diff
@@ -34,7 +34,6 @@
     "* Kernel Density Estimation Plots\n",
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <p>Challenge, first letter always upper case</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #1.    def label_encode(df, column_name, encoded_column_name):
</pre><p>The second parameter is the column name to encode.</p><p>The 3rd parameter is the new column name for the encoded result.</p><p> </p><p>Please fix the documentation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #8.    def create_assert_df():
</pre><p>expected_result = pd.DataFrame(...) You don't need to create a function</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #18.        def test_label_encode_happy_case(self):
</pre><p>You can add more test cases in next PR, such as test_df is none, second/3rd parameter is None or empty string, none value existing in c1, etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #22.            assert_df["c2"] = pd.DataFrame(assert_df["c2"], dtype=int)
</pre><p>why create a df again? you can use the assert df directly</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #25.            actual_result = label_encode(test_df, "c1", "c2")
</pre><p>Please watch my course video to understand how the <code>lable_encode</code> works. Why the c2 is passed into?   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #28.            assert test_df["c2"].equals(assert_df["c2"])
</pre><p>c2 should not be changed. Why you encode c1 but c2 gets updated?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #29.            # assert_frame_equal(test_df,assert_df)
</pre><p>Please always remove the unused code. I have left such comments multiple times.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: <pre class="ql-syntax" spellcheck="false">Line #11.    def create_assert_df():
</pre><p>same as above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: Thanks for reminding

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: ok

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: DONE

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: Finished

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: ok

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: DONE

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:34]: DONE

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 2]
```diff
@@ -33,8 +33,7 @@
     "* Scatter Plots, subplots\n",
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:33]: <pre class="ql-syntax" spellcheck="false">Line #13.            actual_result = label_encode(test_df, "c1", "c1")
</pre><p>give a different name, e.g. encoded_c1, for the new column name.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:33]: <pre class="ql-syntax" spellcheck="false">Line #7.    def clean_fare(df, ticket_price):
</pre><p>Please add some documentation to the clean fare method. Similar to <code>label_encode</code>.</p><p> </p><p>We can actually make this function more general like this and test against it, <code>def impute_with_mean(df, column_name, imputed_column_name):</code></p><p> </p><p>Then you can call it for price as this,  <code>impute_with_mean(df, 'price', 'price_add_average')</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>
